code/modules/vehicles/mecha/_mecha.dm

```diff
@@ -274,8 +274,9 @@
 		return ..()
 	if(phase_state)
 		flick(phase_state, src)
-	var/area/destination_area = get_step(loc, movement_dir).loc
-	if(destination_area.area_flags & NOTELEPORT)
+	var/turf/destination_turf = get_step(loc, movement_dir)
+	var/area/destination_area = destination_turf.loc
+	if(destination_area.area_flags & NOTELEPORT || SSmapping.level_trait(destination_turf.z, ZTRAIT_SECRET))
```

Comment:
have this be as some different zlevel trait instead, there may be times when it is useful to prevent phasing for an entire zlevel, don't know but I don't think it needs to be exclusive to the secret ztrait.

Reply:
There may also be times when you might want an away mission to be jauntable.
Maybe apply this to non-secret away missions too.